README.markdown

```diff
@@ -214,19 +215,94 @@ The key takeaways should be that:
 
 > Watch the tomcat logs to see that the client extension deployed.
 
-To whitness that the microservice will not allow unauthorized requests run the following curl command in a separate terminal while the microservice is running:
+To witness that the microservice will not allow unauthorized requests run the following curl command in a separate terminal while the microservice is running:
 
 ```bash
 curl -v -X POST http://localhost:58081/ticket/object/action/documentation/referral
 ```
 
-Note the response returns an error.
+> Note the response returns an error.
 
 Finally, return to the [main page of our site](http://localhost:8080) and click the `Generate a New Ticket` button. Review the outcome and verify that:
 
 1. a ticket was created
 1. the documentation referrals are added
 
-We've acheived our third business requirement: **Implement Algorithmic Documentation Referral**.
+We've acheived our third business requirement: **Implement Programmatic Documentation Referral**.
 
 Try making other changes to the projects and redeploying the changes. In the case of the microservice make sure not only to execute the deploy task but also to restart it after any changes.
+
+## Ticket cleanup with cron job (Extra credit)
+
+Now that we can create tickets, at some point, we need to clean them up. Let's create a cron job that will delete all tickets that are marked 'done' or 'duplicate'.  To do this we will use a spring boot application that when executed, it will connect to DXP using the generated headless/REST API for ticket objects using another type of client extension `type: oAuthApplicationHeadlessServer`.  This type of OAuth2 application is using the client credentials flow and is not associated with a user but a service account.  One thing that we need to know is that client credential flow in OAuth2 require both a client_id and client_secret, so there will be some additional steps to perform in order to get this working locally.
```

Comment:
Since the _service account_ hasn't been integrated yet, maybe for now we should replace

> is not associated with a user but a service account

by

> is associated with a special account defined for this purpose (the current default uses the instance admin)